my-sms/config/routes.rb

```diff
@@ -1,9 +1,22 @@
 # frozen_string_literal: true
 
 MySms::Application.routes.draw do
-  resources :students
+  get 'comments/create'
+
+  get 'comments/update'
+
+  get 'comments/destroy'
+
+  def note_resource(**kwargs)
+    resources(:notes, notable_type: parent_resource.name.classify, only: %i[index create update destroy], **kwargs)
+  end
+
+  resources :students do
+    note_resource
+  end
```

Comment:
This is a nested notes resource, I just add some extra params to the route because the note association is polymorphic and will not pass the student params as a notable.

The standard way to handle this problem in the Rails community is to add hidden fields to the form AFAICT but I thought it was better separation and more secure to handle it in the routing like this.

Reply:
Hmm I see...

What I would do if I wanted separate CRUD actions for the note is to do the following,

In routes, I would create custom routes for each notes section (could move this to a route concern- not sure if rails 3 supports this)

```ruby
resources :students do
  resources :notes, controller: "students/notes"
end
```

then create a controller `app/controllers/students/notes_controller.rb`, this will be a small file which should include a _controller concern_ for handing the note update and some private methods for finding the notable object. Something like:

```ruby
class Students::NotesController < ...
  include Concerns::NotesConcern

  private

  def find_notable
    @notable = Student.find(...) 
  end

  # probably not necessary if the notables all have the same name for the has_many
  def new_note
    @notable.notes.new
  end
end
```

Then I would move all the code from the notes controller (except the notable finding logic) into the concern placed in `app/controllers/concerns/notes_concern.rb`

Reply:
This is not necessary anymore as I've moved to using `accepts_nested_attributes_for` to perform the note creation logic on the parent resource as you suggested.

That said, rather than separate CRUD actions for the note, what I was actually trying to achieve was to pass a request parameter with the `notable_type` when accessing the nested note resource. By default, Rails would only pass a `student_id` in the params which is fine if your nested resource is not polymorphic.

Reply:
Ah, so you were trying to match all `{notable_type}/{id}/notes/...` routes to the notes controller? for all valid `notable_type` values?

I think in those cases, even though the model is polymorphic, I would still prefer to be explicit about the routes and controllers, and just use concerns or plain ruby modules to extract the shared logic.

Reply:
Yes but not by editing the actual routes. I was going to explicitly define the nested `note_resource` on each of Student, Course and Institute. All I was doing was passing in an extra request param of 'notable_type' so that the controller does not have to have a series of if/else statements to test if `student_id` exists, then `course_id`, etc.